graph/src/data/subgraph/mod.rs

```diff
@@ -660,24 +685,25 @@ impl UnresolvedDataSource {
         } = self;
 
         info!(logger, "Resolve data source"; "name" => &name, "source" => &source.start_block);
-        mapping
-            .resolve(resolver, logger.clone())
-            .join(
-                stream::futures_ordered(
-                    templates
-                        .into_iter()
-                        .map(|template| template.resolve(resolver, logger.clone())),
-                )
-                .collect(),
-            )
-            .map(|(mapping, templates)| DataSource {
-                kind,
-                network,
-                name,
-                source,
-                mapping,
-                templates,
-            })
+
+        let (mapping, templates) = try_join(
+            mapping.resolve(&*resolver, logger),
+            collect_futures(
+                templates
+                    .into_iter()
+                    .map(|template| template.resolve(resolver, logger)),
+            ),
```

Comment:
Here, too, I think we can get away with unordered futures (and hence, a small speed-up).

Reply:
Nevermind (see earlier discussion about the same thing).